File: benchmarks/fibonacci/Fibonacci.java
public class Fibonacci {
    // Configuration
    private static final int N1 = 10;
    private static final int N2 = 20;
    private static final int N3 = 30;
    private static final int N4 = 35;

    // Markers matching Lamina version
    private static final long HEADER_MARKER = 123456789L;
    private static final long FOOTER_MARKER = 987654321L;

    public static long fibonacciIterative(int n) {
        if (n == 0) return 0;
        if (n == 1) return 1;

        long a = 0, b = 1;
        for (int i = 2; i <= n; i++) {
            long temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }

    public static void main(String[] args) {
        // Print header marker
        System.out.println(HEADER_MARKER);

        // Compute and print fibonacci numbers
        System.out.println(fibonacciIterative(N1));
        System.out.println(fibonacciIterative(N2));
        System.out.println(fibonacciIterative(N3));
        System.out.println(fibonacciIterative(N4));

        // Print footer marker
        System.out.println(FOOTER_MARKER);
    }
}

File: benchmarks/fibonacci/fibonacci.cpp
#include <iostream>

// Configuration
#define N1 10
#define N2 20
#define N3 30
#define N4 35

// Markers matching Lamina version
#define HEADER_MARKER 123456789LL
#define FOOTER_MARKER 987654321LL

long long fibonacci_iterative(int n) {
    if (n == 0) return 0;
    if (n == 1) return 1;

    long long a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        long long temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

int main() {
    // Print header marker
    std::cout << HEADER_MARKER << std::endl;

    // Compute and print fibonacci numbers
    std::cout << fibonacci_iterative(N1) << std::endl;
    std::cout << fibonacci_iterative(N2) << std::endl;
    std::cout << fibonacci_iterative(N3) << std::endl;
    std::cout << fibonacci_iterative(N4) << std::endl;

    // Print footer marker
    std::cout << FOOTER_MARKER << std::endl;

    return 0;
}

